Funcky.Async/Extensions/AsyncEnumerableExtensions/AdjacentGroupBy.cs

```diff
@@ -0,0 +1,187 @@
+        /// <typeparam name="TKey">The type of the key returned by keySelector.</typeparam>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="keySelector">A function to extract the key for each element.</param>
+        /// <returns>An IEnumerable{IGrouping{TKey, TSource}}  where each IGrouping{TKey,TElement} object contains a sequence of objects and a key.</returns>
+        [Pure]
+        public static IAsyncEnumerable<IAsyncGrouping<TKey, TSource>> AdjacentGroupBy<TSource, TKey>(
+            this IAsyncEnumerable<TSource> source,
+            Func<TSource, TKey> keySelector)
+            => AdjacentGroupBy(source, keySelector, Identity, CreateGrouping, EqualityComparer<TKey>.Default);
+
+        /// <summary>
+        /// Groups adjacent elements of a source sequence according to a specified key selector function and compares the keys by using a specified comparer.
+        /// </summary>
+        /// <typeparam name="TSource">Type of the elements in <paramref name="source"/> sequence.</typeparam>
+        /// <typeparam name="TKey">The type of the key returned by keySelector.</typeparam>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="keySelector">A function to extract the key for each element.</param>
+        /// <param name="comparer">An IEqualityComparer{T} to compare keys.</param>
+        /// <returns>An IEnumerable{IGrouping{TKey, TSource}}  where each IGrouping{TKey,TElement} object contains a sequence of objects and a key.</returns>
+        [Pure]
+        public static IAsyncEnumerable<IAsyncGrouping<TKey, TSource>> AdjacentGroupBy<TSource, TKey>(
+            this IAsyncEnumerable<TSource> source,
+            Func<TSource, TKey> keySelector,
+            IEqualityComparer<TKey> comparer)
+            => AdjacentGroupBy(source, keySelector, Identity, CreateGrouping, comparer);
+
+        /// <summary>
+        /// Groups adjacent elements of a source sequence according to a key selector function. The keys are compared by using a comparer and each group's elements are projected by using a specified function.
+        /// </summary>
+        /// <typeparam name="TSource">Type of the elements in <paramref name="source"/> sequence.</typeparam>
+        /// <typeparam name="TKey">The type of the key returned by keySelector.</typeparam>
+        /// <typeparam name="TElement">The type of the elements in each IGrouping{TKey,TElement}.</typeparam>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="keySelector">A function to extract the key for each element.</param>
+        /// <param name="elementSelector">A function to map each source element to an element in the IGrouping{TKey,TElement}.</param>
+        /// <returns>An IEnumerable{IGrouping{TKey, TSource}}  where each IGrouping{TKey,TElement} object contains a sequence of objects and a key.</returns>
```

Comment:
Why is the `IEnumerable{IGrouping{TKey, TSource}}` not in a `<see cref="...">`?

Reply:
Fixed in ea1719fee159f0ab827b55e77ff403379a0f94ff